src/core/component.ts

```diff
@@ -67,7 +67,8 @@ const BaseComponent: VueConstructor = Vue.extend({
             const result: Record<string, any> = {
                 integrationOptions:  {
                     watchMethod: this.$_getWatchMethod(),
-                }
+                },
+                ...this.$_getOnInitializingCallback(),
```

Comment:
```suggestion
                ...this.$_extraIntegrationOptions,
```
What about making it just a field?